src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -135,25 +164,213 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
+            if (validTopicWriteAcl == false)
+                msgBuilder.Append($"Topic '{topicName}' : WRITE operation");
+
+            message = msgBuilder.ToString();
+
+        }
+        catch (Exception)
+        {
+            return false;
         }
 
-        return count;
+        return true;
+    }
+
+    private static bool AssertAclMatchingName(string expectedName, AclBinding aclBinding)
+    {
+        return aclBinding.Pattern.ResourcePatternType switch
+        {
+            ResourcePatternType.Literal => IsMatchingLiterally(aclBinding.Pattern.Name, expectedName),
+            ResourcePatternType.Prefixed => IsPrefixMatching(aclBinding.Pattern.Name, expectedName),
+            ResourcePatternType.Match => Regex.IsMatch(aclBinding.Pattern.Name, expectedName),
+            _ => throw new NotSupportedException(
+                $"Resource pattern type '{aclBinding.Pattern.ResourcePatternType}' is not supported.{Environment.NewLine}" +
+                $"Supported resource types are: '{ResourcePatternType.Literal}' and {ResourcePatternType.Prefixed}")
+        };
+    }
+    
+    private static bool IsMatchingLiterally(string pattern, string expected)
+    {
+        if (pattern.Contains('*') == false) 
+            return pattern == expected;
+        
+        string patternRegex = "^" + Regex.Escape(pattern).Replace("\\*", ".*") + "$";
+        return Regex.IsMatch(expected, patternRegex);
+    }
+    
+    private static bool IsPrefixMatching(string pattern, string expected)
+    {
+        if (pattern.Contains('*') == false)
+            return expected.StartsWith(pattern);
+        
+        // If the pattern contains '*', use wildcard matching in the prefix
+        string[] parts = pattern.Split('*');
+        
+        // if pattern is '*', it's a match
+        if (parts.Length == 0) 
+            return true; 
+        
+        int index = 0; // try to find part after part in the expected string
+        foreach (var substring in parts)
+        {
+            index = expected.IndexOf(substring, index, StringComparison.Ordinal);
+            if (index == -1) 
+                return false; // If a part is not found in order, it's not a match
+            index += substring.Length;
+        }
+        return true;
+    }
+    
+
+    private void ThrowTransactionalIdAclAuthorizationException(Exception e)
+    {
+        string msg = $"ETL process: {Name}. Failed to initialize transactions for the producer instance. " +
+                     $"Add an required ACL permissions to your API Key that will make WRITE and DESCRIBE operations possible for transactional ID '{TransactionalId}'. ";
+        throw new QueueLoadException(msg, e);
+    }
+
+    private bool TryAbortTransactionAfterException(StringBuilder errorMessageBuilder, out Exception abortException)
+    {
+        abortException = null;
+        try
+        {
+            errorMessageBuilder.Append($"{Environment.NewLine}Aborting the transaction. ");
```

Comment:
Is important to add this the exception message even if we succeed in aborting? It looks a bit strange that we pass `StringBuilder errorMessageBuilder` here. Shouldn't we handle the message externally based on value returned from this method?

```
StringBuilder errorMessageBuilder = ....


if (TryAbortTransactionAfterException(out _) == false)
{
   errorMessageBuilder.Append("Aborting Kafka transaction failed too.");
}
```

or if you care about exception:

```
if (TryAbortTransactionAfterException(out var abortException) == false)
{
   // do something with abortException
}
```

Reply:
done